### DISCUSSION TITLE
Does the schema parser used by vscode-yaml idiosyncratically parse `$refs` for `patternProperties` elements?

### DISCUSSION THREAD

**Discussion by jglev** [redhat-developer/vscode-yaml#910] [Q&A]
Hello,

First, thank you very much for your work on `vscode-yaml`! I'm very grateful to be able to use it!

I have the following schema, which uses a `$ref` to refer to a `patternProperties` element within a [remote schema](https://raw.githubusercontent.com/kedro-org/kedro/develop/static/jsonschema/kedro-catalog-0.18.json):

```json
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "type": "object",
  "patternProperties": {
    "^[a-z0-9-_]+$": {
      "required": [
        "type"
      ],
      "properties": {
        "type": {
          "oneOf": [
            {
              "$ref": "https://raw.githubusercontent.com/kedro-org/kedro/develop/static/jsonschema/kedro-catalog-0.18.json#/patternProperties/^[a-z0-9-_]+$/properties/type/"
            },
            {
              "type": "string",
              "enum": [
                "Test2"
              ]
            }
          ]
        }
      }
    }
  }
}
```

I then am using `vscode-yaml` to validate the following YAML file:

```yaml
example1:
  type: pandas.CSVDataSet  # An enum value from the referenced schema
  filepath: 'path/to/file1'


example2:
  type: Test2  # An enum value from the new schema
  filepath: 'path/to/file2'
```

I see an error in VSCode when viewing this file: `$ref '/patternProperties/^[a-z0-9-_]+$/properties/type/' in 'https://raw.githubusercontent.com/kedro-org/kedro/develop/static/jsonschema/kedro-catalog-0.18.json' can not be resolved.`

This schema and content _seem_ to be valid, to my understanding: Entering the schema above and the equivalent JSON into [this JSON Schema Validator tool](https://www.jsonschemavalidator.net/) reports no errors.

```json
{
  "example": {
    "type": "pandas.CSVDataSet",
    "filepath": "path/to/file1"
  },	
  "example2": {
    "type": "Test2",
    "filepath": "path/to/file2"
  }
}
```

Is the schema parser used by `vscode-yaml` failing to resolve a valid `$ref`? Or does something in the example content above look incorrect? I would be grateful for any advice!

**Comment by gorkem** [comment]
Could this be related to [this issue](https://github.com/redhat-developer/yaml-language-server/issues/823)
